tools/generators/legacy/src/Generator/DisambiguateTargets.swift

```diff
@@ -76,12 +76,19 @@ extension Generator {
             guard labelsByModuleNameAndProductType[moduleNameAndProductType]!
                 .count != 1
             else {
+                let baseName: String
+                switch consolidatedTargets.targetNameMode {
+                case .auto:
+                    baseName = target.moduleName
+                case .label:
+                    baseName = "\(target.label)"
+                }
                 uniqueValues[key] = DisambiguatedTarget(
                     name: names[moduleNameAndProductType.normalizedModuleName]!
                         .uniqueName(
                             for: target,
                             key: key,
-                            baseName: target.moduleName
+                            baseName: baseName
```

Comment:
I think instead of this, we can have lines 74-104 (in the original source code) only happen when `consolidatedTargets.targetNameMode == .auto`. If you do that, it will reduce the amount of work, and do 106-113, which is what you are trying to do anyway.

Reply:
I used a switch instead because it will force to update it if a new option is added, but the downside of this is that some small part is duplicated. I hope you don't mind.

I also populate the label dictionary for each target (previously it was skipped due to `continue` inside guard statements above), which is perhaps even more appropriate, but has some memory penalty ( not big, I think ). This was necessary because the `target.normalizedLabel` did not exist in `labels` dictionary if the label option was selected.